int32 data_num
---
bool finish
int32 concept_num